sentence.go

```diff
@@ -74,69 +79,137 @@ func (s BaseSentence) TalkerID() string {
 // String formats the sentence into a string
 func (s BaseSentence) String() string { return s.Raw }
 
-// parseSentence parses a raw message into it's fields
-func parseSentence(raw string) (BaseSentence, error) {
+// SentenceParser is configurable parser instance to parse raw input into NMEA0183 Sentence
+//
+// SentenceParser fields/methods are not co-routine safe!
+type SentenceParser struct {
+	// CustomParsers allows registering additional parsers
+	CustomParsers map[string]ParserFunc
+
+	// ParsePrefix takes in the sentence first field (NMEA0183 address) and splits it into a talker id and sentence type
+	ParsePrefix func(prefix string) (talkerID string, sentence string, err error)
+
+	// CheckCRC allows custom handling of checksum checking based on parsed sentence
+	CheckCRC func(sentence BaseSentence, rawFields string) error
+
+	// OnTagBlock is callback to handle all parsed tag blocks even for lines containing only a tag block and
+	// allows to track multiline tag group separate lines. Logic how to combine/assemble multiline tag group
+	// should be implemented outside SentenceParser
+	//
+	// Example of group of 3:
+	// \g:1-3-1234,s:r3669961,c:1120959341*hh\
+	// \g:2-3-1234*hh\!ABVDM,1,1,1,B,.....,0*hh
+	// \g:3-3-1234*hh\$ABVSI,r3669961,1,013536.96326433,1386,-98,,*hh
+	OnTagBlock func(tagBlock TagBlock)
```

Comment:
Do you think we need an error return here?

Reply:
I think my though was here that failure to parse tag block should still allow parsing sentence but  if no error is returned this is still possible.  I'll add error as return value.